orthoinference/src/main/java/org/reactome/orthoinference/ReactionInferrer.java

```diff
@@ -52,87 +51,72 @@ public static void inferReaction(GKInstance reactionInst) throws Exception
-							dba.updateInstanceAttribute(reactionInst, orthologousEvent);
-							
-							inferredEvent.put(reactionInst, infReactionInst);
-							
-							// Regulations instances require the DB to contain the inferred ReactionlikeEvent, so Regulations inference happens post-inference
-							if (inferredRegulations.size() > 0)
+						if (infReactionInst.getSchemClass().isValidAttribute(inferredFrom))
+						{
+							infReactionInst = InstanceUtilities.addAttributeValueIfNecessary(infReactionInst, reactionInst, inferredFrom);
+							dba.updateInstanceAttribute(infReactionInst, inferredFrom);
+						}
+						infReactionInst = InstanceUtilities.addAttributeValueIfNecessary(infReactionInst, reactionInst, orthologousEvent);
+						dba.updateInstanceAttribute(infReactionInst, orthologousEvent);
+						reactionInst.addAttributeValue(orthologousEvent, infReactionInst);
+						dba.updateInstanceAttribute(reactionInst, orthologousEvent);
+
+						inferredEvent.put(reactionInst, infReactionInst);
+
+						// Regulations instances require the DB to contain the inferred ReactionlikeEvent, so Regulations inference happens post-inference
+						if (inferredRegulations.size() > 0)
+						{
+							logger.info("Number of regulator(s) inferred: " + inferredRegulations.size());
+							for (GKInstance infRegulation : inferredRegulations)
 							{
-								logger.info("Number of regulator(s) inferred: " + inferredRegulations.size());
-								for (GKInstance infRegulation : inferredRegulations)
-								{
-									infRegulation = InstanceUtilities.checkForIdenticalInstances(infRegulation, null);
-									infReactionInst.addAttributeValue("regulatedBy", infRegulation);
-									dba.updateInstanceAttribute(infReactionInst, "regulatedBy");
-								}
+								infRegulation = InstanceUtilities.checkForIdenticalInstances(infRegulation, null);
+								infReactionInst.addAttributeValue("regulatedBy", infRegulation);
+								dba.updateInstanceAttribute(infReactionInst, "regulatedBy");
 							}
-							// After successfully adding a new inferred instance to the DB, it is recorded in the 'inferred reactions' file
-							inferredCount++;
-							inferrableHumanEvents.add(reactionInst);
-							String inferredEvent = infReactionInst.getAttributeValue(DB_ID).toString() + "\t" + infReactionInst.getDisplayName() + "\n";	
-							Files.write(Paths.get(inferredFilehandle), inferredEvent.getBytes(), StandardOpenOption.APPEND);
-						} else {
-							logger.info("Catalyst inference unsuccessful -- terminating inference for " + reactionInst);
 						}
+						// After successfully adding a new inferred instance to the DB, it is recorded in the 'inferred reactions' file
+						inferredCount++;
+						inferrableHumanEvents.add(reactionInst);
+						String inferredEvent = infReactionInst.getAttributeValue(DB_ID).toString() + "\t" + infReactionInst.getDisplayName() + "\n";
+						Files.write(Paths.get(inferredFilehandle), inferredEvent.getBytes(), StandardOpenOption.APPEND);
```

Comment:
If the reaction db_id and display name are needed, the `getExtendedDisplayName()` method could be used in place of the `inferredEvent` temp variable:

```
Files.write(
    Paths.get(inferredFilehandle), 
    infReactionInst.getExtendedDisplayName().getBytes(), 
    StandardOpenOption.APPEND
);
```